src/components/extensive/Tables/TreeSpeciesTable.tsx

```diff
@@ -37,11 +43,42 @@ const TreeSpeciesTable = ({ modelName, modelUUID, collection, onFetch }: TreeSpe
       ? treeSpecies?.data?.reduce((total, item) => total + (typeof item.amount === "number" ? 1 : 0), 0) > 0
       : false;
 
+  const { data: record } = useGetV2FormsENTITYUUID<{ data: GetV2FormsENTITYUUIDResponse }>({
+    pathParams: {
+      uuid: modelUUID,
+      entity: modelName
+    }
+  });
+
+  const viewDataDisturbances = record?.data?.form?.form_sections.map((questions: any) =>
+    questions.form_questions.map((item: any) => item.uuid).map((item: any) => record?.data?.answers?.[item])
+  );
+
+  const verifydata = () => {
+    record?.data?.form?.form_sections.forEach((section: any, sectionIndex: number) => {
+      section.form_questions.forEach((question: any, questionIndex: number) => {
+        if (question.children) {
+          question.children.forEach((child: any) => {
+            if (child.input_type === "treeSpecies") {
+              setShowTreeSpecies(viewDataDisturbances?.[sectionIndex as number]?.[questionIndex as number]);
+              viewDataDisturbances?.[sectionIndex as number]?.[questionIndex as number];
+            }
+          });
+        }
+      });
+    });
+    return false;
+  };
+
+  useEffect(() => {
+    verifydata();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [record]);
```

Comment:
now that I see that you are duplicating the code, I think is better to have a custom hook for the function above